tests/cli/workflow/nf_analysis/test_cli_compound_commands.py

```diff
@@ -0,0 +1,115 @@
+    request.getfixturevalue(f"{workflow}_mock_analysis_finish")
+
+    # GIVEN a mocked deliverables template
+    mocker.patch.object(
+        NfAnalysisAPI,
+        "get_deliverables_template_content",
+        return_value=deliverables_template_content,
+    )
+
+    # Set Housekeeper to an empty real Housekeeper store
+    context.housekeeper_api_: HousekeeperAPI = real_housekeeper_api
+    context.meta_apis["analysis_api"].housekeeper_api = real_housekeeper_api
+
+    # Make sure the bundle was not present in hk
+    assert not context.housekeeper_api.bundle(case_id)
+
+    # Make sure analysis not already stored in status_db
+    assert not context.status_db.get_case_by_internal_id(internal_id=case_id).analyses
+
+    # GIVEN that HermesAPI returns a deliverables output
+    mocker.patch.object(HermesApi, "convert_deliverables")
+    HermesApi.convert_deliverables.return_value = CGDeliverables(**hermes_deliverables)
+
+    # WHEN running command
+    result = cli_runner.invoke(workflow_cli, [workflow, "store", case_id], obj=context)
+
+    # THEN bundle should be successfully added to Housekeeper and StatusDB
+    assert result.exit_code == EXIT_SUCCESS
+    assert "Analysis successfully stored in Housekeeper" in caplog.text
+    assert "Analysis successfully stored in StatusDB" in caplog.text
+    assert context.status_db.get_case_by_internal_id(internal_id=case_id).analyses
+    assert context.housekeeper_api.bundle(case_id)
+
+
+@pytest.mark.parametrize(
+    "workflow",
+    [Workflow.RNAFUSION, Workflow.TAXPROFILER],
+)
+def test_store_fail(
+    cli_runner: CliRunner,
+    workflow: Workflow,
+    rnafusion_context: CGConfig,
+    real_housekeeper_api: HousekeeperAPI,
+    deliverables_template_content: list[dict],
+    caplog: LogCaptureFixture,
+    request: FixtureRequest,
+    mocker,
+):
+    """Test store command fails when a case did not finish for a workflow."""
+    caplog.set_level(logging.INFO)
+
+    # GIVEN each fixture is being initialised
+    context: CGConfig = request.getfixturevalue(f"{workflow}_context")
+
+    # GIVEN CASE ID where analysis finish is not mocked
+    case_id_fail: str = request.getfixturevalue(f"{workflow}_case_id")
+
+    # GIVEN a mocked deliverables template
+    mocker.patch.object(
+        NfAnalysisAPI,
+        "get_deliverables_template_content",
+        return_value=deliverables_template_content,
+    )
+
+    # Set Housekeeper to an empty real Housekeeper store
+    context.housekeeper_api_: HousekeeperAPI = real_housekeeper_api
+    context.meta_apis["analysis_api"].housekeeper_api = real_housekeeper_api
+
+    # Make sure the bundle was not present in hk
+    assert not context.housekeeper_api.bundle(case_id_fail)
+
+    # WHEN running command
+    result_fail = cli_runner.invoke(workflow_cli, [workflow, "store", case_id_fail], obj=context)
+
+    # THEN bundle exist status should be
+    assert result_fail.exit_code != EXIT_SUCCESS
```

Comment:
WWould be nice to assert that the bundle was created.

Reply:
@henrikstranneheim Updated the tests.